core/types/interoptypes/interop.go

```diff
@@ -188,5 +191,23 @@ func (ed *ExecutingDescriptor) UnmarshalJSON(input []byte) error {
 		return err
 	}
 	ed.Timestamp = uint64(dec.Timestamp)
+	ed.Timeout = uint64(dec.Timeout)
 	return nil
 }
+
+func TxToInteropAccessList(tx *types.Transaction) []common.Hash {
+	if tx == nil {
+		return nil
+	}
+	al := tx.AccessList()
+	if len(al) == 0 {
+		return nil
+	}
+	var hashes []common.Hash
+	for i := range al {
+		if al[i].Address == params.InteropCrossL2InboxAddress {
```

Comment:
i think you would have only 1 item with InteropCrossL2InboxAddress so you could return early

Reply:
Thanks for the review @SozinM!

Regarding only having 1 item with InteropCrossL2InboxAddress -- [EIP2930](https://eips.ethereum.org/EIPS/eip-2930) specifies duplicates are allowed, so theoretically a caller could specify the same address multiple times, yeah?